Net/src/SocketImpl.cpp

```diff
@@ -104,7 +104,7 @@ SocketImpl* SocketImpl::acceptConnection(SocketAddress& clientAddr)
 {
 	if (_sockfd == POCO_INVALID_SOCKET) throw InvalidSocketException();
 
-	char buffer[SocketAddress::MAX_ADDRESS_LENGTH];
+	Poco::Int32 buffer[(SocketAddress::MAX_ADDRESS_LENGTH >> 2) + 1];
```

Comment:
Oh, I didn't see that you created a pull request for the issue #2492 
I think this fix is more a workaround. The real solution should be to use `sockaddr_storage` See pull request #2523